### PR TITLE
feat: retry on 502 error code

### PR DESCRIPTION
**Pull request by astencel-sumo**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,6 +1,46 @@
...
+
+```sh
+bundle install
+```
+
+## Running tests
+
+Some of the tests try to send actual data to an actual Sumo Logic account.
+To run those tests, you need to make `sumo_url` environment variable available.
+If the `sumo_url` environment variable is not present, the tests reaching Sumo Logic will be skipped.
```
Comment (sumo-drosiek) [inline CONTRIBUTING.md:26]: sounds like case for receiver-mock :thinking:

File: vagrant/provision.sh [hunk 2]
```diff
@@ -3,25 +3,39 @@
...
+echo 'eval "$(rbenv init - bash)"' >> ~/.bashrc
+# Install rbenv-build plugin for `rbenv install` command https://github.com/rbenv/ruby-build#installation
+export PATH="$HOME/.rbenv/bin:$PATH"
+eval "$(rbenv init - bash)"
+mkdir -p "$(rbenv root)"/plugins
+git clone https://github.com/rbenv/ruby-build.git "$(rbenv root)"/plugins/ruby-build
+
+# Install Java SE as a prerequisite for JRuby.
+# Use the same version as in Logstash Dockerfile https://github.com/elastic/logstash/blob/37e1db0c129c03cfd7b724775d26a06eb5a1fe39/Dockerfile#L5.
+sudo apt install --yes openjdk-11-jdk-headless
```
Comment (sumo-drosiek) [inline vagrant/provision.sh:36]: Jruby :scream:

**Review by sumo-drosiek — APPROVED** [review]

**Review by pmalek-sumo — APPROVED** [review]
Nicely tidied up here! 🎸

File: CONTRIBUTING.md (hunk 1 above)
Comment (astencel-sumo) [reply CONTRIBUTING.md:26]: Yes, very much. Pull requests are very welcome :D

Comment (sumo-drosiek) [reply CONTRIBUTING.md:26]: Could you create issue for that? :sweat:

Comment (astencel-sumo) [reply CONTRIBUTING.md:26]: https://github.com/SumoLogic/logstash-output-sumologic/issues/69

**Pull request merged**